operators/vision/draw_bounding_box.cc

```diff
@@ -0,0 +1,248 @@
+  [[nodiscard]] int64_t NumBoxes() const {
+    return shape_[0];
+  }
+
+  auto BBoxMode() const {
+    return bbox_mode_;
+  }
+
+ private:
+  const std::vector<int64_t>& shape_;
+  std::vector<gsl::span<const float>> boxes_by_score_;
+  BoundingBoxFormat bbox_mode_;
+};
+
+// Draw a line on the image
+void DrawLineInHorizon(ImageView& image, int64_t x_start, int64_t y_start, int64_t line_length,
+                       gsl::span<const uint8_t> color, int64_t thickness) {
+  // boundary check
+  thickness = std::clamp<int64_t>(thickness, 0, image.height - x_start);
+  auto stride = image.width * image.channels;
+  auto point_start = image.data.begin() + x_start * stride + y_start * image.channels;
+  for (auto start = point_start; start < point_start + line_length * image.channels; start += image.channels) {
+    std::copy_n(color.begin(), color.size(), start);
+  }
+
+  for (int64_t i = 1; i < thickness; ++i) {
+    auto xy_point_start = point_start + i * stride;
+    auto end = xy_point_start + line_length * image.channels;
+    std::copy_n(point_start, line_length * image.channels, xy_point_start);
+  }
+}
+
+void DrawLineInVertical(ImageView& image, int64_t x_start, int64_t y_start, int64_t line_length,
+                        gsl::span<const uint8_t> color, int64_t thickness) {
+  // boundary check
+  thickness = std::clamp<int64_t>(thickness, 0, image.width - y_start);
+  auto stride = image.width * image.channels;
+  auto point_start = image.data.begin() + x_start * stride + y_start * image.channels;
+  auto point_end = point_start + line_length * stride;
+
+  for (int64_t i = 0; i < thickness; ++i) {
+    std::copy_n(color.begin(), color.size(), point_start + i * image.channels);
+  }
+  for (auto xy_point_start = point_start + stride; xy_point_start < point_end; xy_point_start += stride) {
+    std::copy_n(point_start, thickness * image.channels, xy_point_start);
+  }
+}
+
+// Draw a box on the image with given thickness and color
+void DrawBox(ImageView& image, gsl::span<const float> box, BoundingBoxFormat bbox_mode,
+             gsl::span<const uint8_t> color, int64_t thickness) {
+  // -------(1)------
+  // |              |
+  //(2)            (4)
+  // |              |
+  // -------(3)------
+  float point_x1 = box[0];
+  float point_x2 = box[1];
+  float point_x3 = box[2];
+  float point_x4 = box[3];
+  if (bbox_mode == BoundingBoxFormat::Center_XYWH) {
+    point_x1 = box[0] - point_x3 / 2;
+    point_x2 = box[1] - point_x4 / 2;
+    point_x3 = box[0] + point_x3 / 2;
+    point_x4 = box[1] + point_x4 / 2;
+  } else if (bbox_mode == BoundingBoxFormat::XYWH) {
+    point_x1 = box[0];
+    point_x2 = box[1];
+    point_x3 = box[0] + box[2];
+    point_x4 = box[1] + box[3];
+  }
+  int64_t x_start = static_cast<int64_t>(std::rintf(point_x1));
+  int64_t y_start = static_cast<int64_t>(std::rintf(point_x2));
+  int64_t x_end = static_cast<int64_t>(std::rintf(point_x3));
+  int64_t y_end = static_cast<int64_t>(std::rintf(point_x4));
```

Comment:
Why rintf? Doesn't that make it less deterministic than `round` as the current rounding mode is used, and it could be harder to debug an issue?